### ISSUE TITLE
Node Socket timeout

### ISSUE THREAD

**Issue by pierre-elie** [evernote/evernote-sdk-js#34]
Created node HTTP requests [do not have a timeout set](https://github.com/evernote/evernote-sdk-js/blob/master/evernote-sdk-js/thrift/lib/node/thrift-node-binary.js#L108-143) in the thrift lib.
By default, [sockets do not timeout](http://nodejs.org/api/net.html#net_socket_settimeout_timeout_callback).

I think it wouldn’t hurt setting a reasonable value for these.

**Comment by install** [comment]
Hi Pierre,
I've added a request timeout. 
see: https://github.com/evernote/evernote-sdk-js/commit/725f38f4d83c355a80fec2592ea08f414e70b917#diff-2cbdfef8dfdd7a5ed197280ed7b97d3aR147

Thanks for your suggestion and let me know if this is satisfactory.

**Comment by pierre-elie** [comment]
Thanks @install, will give it a try right away

**Comment by pierre-elie** [comment]
It turns out the callback is called twice in case of a timeout.

Here’s a quick example, simulating a timeout with your code. 
I basically replaced `callback(...)` by `console.log('Calling callback (...)')`.

```js
var http = require('http');

http.createServer(function (req, res) {
  console.log('Received Request');
  setTimeout(function() {
    res.writeHead(200);
    res.end('pong');
  }, 550);
}).listen(8080, function() {
  var options = {
    hostname: 'localhost',
    port: 8080,
    path: '/',
    method: 'GET'
  };

  var req = http.request(options, function(res) {
    res.on('end', function() {
      console.log('Calling callback (end)');
    });
  });

  req.on('error', function(e) {
    console.log('Calling callback (error)', e);
  });

  req.on('socket', function(socket) {
    socket.setTimeout(500);
    socket.on('timeout', function() {
      console.log('Calling callback (timeout)');
      req.abort();
    });
  });

  req.end();
});
```

__Output:__
```
Received Request
Calling callback (timeout)
Calling callback (error) { [Error: socket hang up] code: 'ECONNRESET' }
```

**Comment by pierre-elie** [comment]
Also, could the timeout be configurable?
Or at least be less than 30 seconds, which seems too high.

**Comment by install** [comment]
I removed the callback from the socket timeout function because it should emit an error if the socket is closed before it receives a response.  This change is available in  `1.25.8-beta` version of the npm module.
